vendor/zkllvm-metacraft-circuits/src/circuit_input_generators/serialization/json_serialization_utils.h

```diff
@@ -0,0 +1,117 @@
+#pragma once
+
+#include <sstream>
+
+#include "json/json.hpp"
+using namespace nlohmann;
+
+nlohmann::json pack_int_json(uint64_t val) {
+    nlohmann::json j;
+    j["int"] = val;
+    return j;
+}
+
+std::ostream& operator<<(std::ostream& dest, __uint128_t value) {
+    std::ostream::sentry s(dest);
+    if (s) {
+        char buffer[128];
+        char* d = std::end(buffer);
+        do {
+            --d;
+            *d = "0123456789"[value % 10];
```

Comment:
value % 10 + '0'. Rename value to last_digit

Reply:
The value contains all the digits, we are taking them one by one. I have added comments.

Reply:
Excuse me, I meant to say to assign value % 10 to a variable last_digit that is value % 10 + '0'